src/v/transform/rpc/client.cc

```diff
@@ -855,4 +856,86 @@ ss::future<> client::update_wasm_binary_size() {
       cluster::error_category().message(int(ec)));
 }
 
+ss::future<result<model::transform_offsets_map, cluster::errc>>
+client::list_committed_offsets() {
+    auto cfg = _topic_metadata->find_topic_cfg(model::transform_offsets_nt);
+    if (!cfg) {
+        co_return model::transform_offsets_map{};
+    }
+    using ret_t = result<model::transform_offsets_map, cluster::errc>;
+    co_return co_await ss::map_reduce(
+      boost::irange(0, cfg->partition_count),
+      [this](int32_t id) {
+          auto partition = model::partition_id(id);
```

Comment:
it's really unsatisfying to my brain that we are generating the partition ids here rather than having the set of partitions available to iterate over. is that information not available here, or is this code fundamentally acting like a client?

Reply:
We only know how many partitions there are and we need to map reduce over all of them. This code is a client fwiw. Is there a way to get a materialized set of all partitions?

Reply:
not sure. but it representing a client fundamentally makes much more sense to me!